whois-nrtm4/src/main/java/net/ripe/db/nrtm4/util/SnapshotUpdateWindow.java

```diff
@@ -0,0 +1,39 @@
+package net.ripe.db.nrtm4.util;
+
+import net.ripe.db.nrtm4.persist.SnapshotFile;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.util.Date;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+@Component
+public class SnapshotUpdateWindow {
+
+    private final Date startTime;
+    private final Date endTime;
+
+    SnapshotUpdateWindow(@Value("${nrtm.snapshot.window:00:00-04:00}") final String snapshotWindowString) {
+        // parse it into a representation that includes midnight -- so 10:00-06:00 starts at 10am and ends at 6am the following day
+        final Pattern PATTERN = Pattern.compile("(\\d{2}):(\\d{2})-(\\d{2}):(\\d{2})");
```

Comment:
Use `java.time.format.DateTimeFormatter`

Reply:
class was removed -- it's just a sketch that i happened to leave in the commit.